### PR TITLE
TDP Finder based on a Kaggle Dataset

### PR DESCRIPTION
**Pull request by gnanakeethan**
### Types of changes

- [ ] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

Adding a TDP Finder based on a Kaggle Dataset found in https://www.kaggle.com/datasets/lincolnzh/cpu-specifications-dataset. A filtered down version of CPU names & TDP is embedded into the IEF via this. Such modified data is permissible under the [CC BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license that originally accompanied the dataset.

### CODE REVIEW

**Comment by jawache** [comment]
@gnanakeethan it's missing a readme file BTW

**Comment by gnanakeethan** [comment]
@jawache I will add documentation and partial matching within today / tomorrow.

**Comment by jmcook1186** [comment]
sgtm overall, but I think it would be good to clarify the success/failure modes for processor names. For example, the following processor names successfully return tdp values:

- `AMD 3020e`
- `Intel Xeon E5-2676 v3`
- `Intel Xeon`

But the following fail:

- `Intel Xeon E5-2676`
- `AMD 3020`

Therefore, it is not necessarily true that adding more detail gives more certainty that the lookup will be successful. For example, `Intel Xeon E5-2676 v3`: remove *only* the `v3` tag and the request fails, but remove the entire `E5-2676 v3` suffix and the request succeeds. What is the ruleset that determines lookup success/failure?

It is also not clear which specific processor, or aggregate metric, is being reported when a fuzzy name successfully returns a tdp (what is actually being reported when the name is `Intel Xeon`?)

I don't think these are necessarily blocking issues, but the detail should at least be constrained and documented.

**Comment by gnanakeethan** [comment]
@jmcook1186 I tried multiple fuzzy finders. It is much complex to implement it in purely Javascript / Typescript environment. 

Seemingly few of them didn't work. 

Best Regards,
Gnanakeethan.

**Comment by jawache** [comment]
@gnanakeethan can you elaborate on why fuzzzy matching doesn't work? What exactly are you trying to do and how isn't it working?

**Comment by gnanakeethan** [comment]
@jmcook1186 @jawache 

We can also use the newly released API Endpoint from Boavizta for this purpose.

**Comment by srini1978** [comment]
> @jmcook1186 @jawache
> 
> We can also use the newly released API Endpoint from Boavizta for this purpose.
> 
> https://api.boavizta.org/docs#/utils/name_to_cpu_v1_utils_name_to_cpu_get

Tried it with a few Azure VMs and it seems to work

**Review by narekhovhannisyan — APPROVED** [review]

**Comment by gnanakeethan** [comment]
@jmcook1186 Changes are finalised. Please review.

File: docs/implementations/tdp-finder.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# TDP Finder Model
+
+## Scope
+
+The TDP Finder model enables finding the TDP of known set processor in the model dataset. Highest TDP value of provided CPUs is provided as the TDP power. The TDP is then used by other models to calculate the e-cpu value.
```
Comment (jmcook1186) [inline docs/implementations/tdp-finder.md:5]:
```suggestion
The TDP Finder model finds the thermal design power (TDP) of a given processor by looking it up in the model datasets. There are scenarios where the lookup can return multiple possible TDP values. In these cases, we return the maximum of the possible values. The TDP is then used by other models to calculate the `energy-cpu` value. 
```

**Review by jmcook1186 — APPROVED** [review]
sgtm

**Pull request merged**